_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -1399,6 +1409,272 @@ module "vpc" {
+
+
+[source,hcl]
+----
+locals {
+  region_vars  = read_terragrunt_config(find_in_parent_folders("region.hcl"))
+  profile_name = "<your AWS profile>"
+  role_arn     = "<your IAM role>"
+  creds        = split(",", chomp(run_cmd("bin/get_aws_creds.py",
+                       "--profile-name",
+                       "${local.profile_name}",
+                       "--role-arn",
+                       "${local.role_arn}")))
+  access_key     = local.creds[0]
+  secret_key     = local.creds[1]
+  session_token  = local.creds[2]
+}
+
+generate "aws_provider" {
+  path      = "provider.tf"
+  if_exists = "overwrite"
+  contents = <<-EOF
+provider "aws" {
+  region     = "${local.region_vars.locals.aws_region}"
+  access_key = "${local.access_key}"
+  secret_key = "${local.secret_key}"
+  token      = "${local.session_token}"
+}
+EOF
+}
+----
+
+In this example, a simple Python script called `get_aws_creds.py` is invoked to assume the role defined in `local.role_arn`. The credentials are returned a comma-separated value of the form `access_key,secret_key,session_token`. The `split()` function separates the values in to a list whose elements are then used in the provider configuration.
+
+In this case, the credentials used by the provider are temporary and not stored in the configuration directly. This avoids the security pitfalls associated with the second approach, and yet still leverages the implicit workspace creation, avoiding the need to set up workspaces and environment variables in advance. If you'd like more information about this configuration, please email mailto:support@gruntwork.io[support@gruntwork.io].
+
+
+===== Summarizing Terragrunt and TFC/TFE compatibility
+
+Using the combination of features above, you can effectively use Terragrunt with TFC or TFE as a remote backend. Terragrunt can generate backend blocks, tfvars files, and provider configurations, resulting in dynamic workspace creation. Commands like `terragrunt apply-all` will operate in the same way as they do without TFC/TFE, calling each module dependency in order, and allowing for passing outputs between modules.
+
+IMPORTANT: The https://gruntwork.io/reference-architecture/[Gruntwork Reference Architecture] is not built for use with TFC or TFE. Although the reference architecture is deployed using Terragrunt, the state is stored in S3, and the configurations discussed above are not included. Migrating from the standard Gruntwork reference architecture to TFC is not supported.
```

Comment:
This makes it sound like migrating to TFC is impossible, when in reality, it's just tedious. To do it, you'd have to update the root `terragrunt.hcl` file(s) to use TFC/TFE as a backend, run `init` in each module, and enter "yes" when prompted if you want to migrate the state. Once you do that, you can use the Ref Arch with TFC/TFE as a remote backend (but still can't trigger things from the UI).

Reply:
My point was that we don't want to support this.